internal/unstructured/unstructured_test.go

```diff
@@ -0,0 +1,33 @@
+package unstructured
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+)
+
+type mockCallback struct {
+	mock.Mock
+}
+
+func (m *mockCallback) VoidMethod(x interface{}) {
+	m.Called(x)
+}
+
+func TestFindAll(t *testing.T) {
```

Comment:
I think this function is simple and pure enough that we can avoid mocks and just have a test case that shows exactly how to use the function:

```go
package unstructured

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestFindAll(t *testing.T) {
	obj := map[string]interface{}{
		"bar": "happy",
		"foo": map[string]interface{}{
			"bar": "path",
		},
		"qux": 2,
	}

	foundValues := []string{}

	FindAll(obj, "bar", func(x interface{}) {
		foundValues = append(foundValues, x.(string))
	})

	assert.ElementsMatch(t, foundValues, []string{"happy", "path"})
}
```